### DISCUSSION TITLE
Search side-panel showing all hits in a document

### DISCUSSION THREAD

**Discussion by giwul70** [sumatrapdfreader/sumatrapdf#4033] [Ideas]
Suggestion: a separate search side panel (enable/disable)  showing all hits within a document, including (parts) of the lines where the searchterm has been found. This way users can quickly jump to the right position in the document.

Something like the below.

![Search Side Panel-SumatraPDF sugestion-18012024](https://github.com/sumatrapdfreader/sumatrapdf/assets/101186455/6a59e681-cbfe-4f13-822a-6071fd6c5fc4)

**Comment by Peter-202122** [comment]
I think this is a good idea.

I just would like to appeal that the "search/find as you type" feature which currently exists in Sumatra will be kept (at least as an option). I mean the feature that already while typing something into the search box, the program jumps automatically to the (first) position of the document where the typed item can be found.

That's a feature which accelerates searching/finding very much (especially if the search term exists only once within the document). It is one of the reasons why I prefer Sumatra to other pdf-readers (I think among them also the one from which you haven taken the screenshot): In such other pdf-readers first you have to type the term you are looking for into the search box, then you have to click on the button "Search" and afterwards you have to click on the first entry in the list of the search panel in order to see the item within the document. No doubt that in some cases this may be sufficient or even useful. But in other cases it is quite cumbersome compared with the method how it works e.g. in Sumatra.

So I would say "Yes" to both features: the suggested search side-panel combined in some way with the existing "search as you type" functionality.

**Reply by giwul70** [reply]
Many thanks for the reply.

Yes, I very much agree that the current search is terrific already: it is extremely fast and there is no waiting to skip to the next hit.
It is really super and beyond discussion. 

As said, this 'side-panel' (optionally enable/disable) with some text before and after the hit will make probably make it easier for users to directly jump to the most likely hit.
OTOH... I do understand that this might mean a major 'overhaul' of the GUI, i.e. and a  _lot_  of programming...
Anyway, 'fingers crossed' :-)

Again, thanks.

**Comment by christopherquenneville** [comment]
I need this aswell

**Comment by PetrAdamovsky** [comment]
I would appreciate this function too. Advanced search after Ctrl+Shift+F pressed.
I use it often in Foxit Reader, because the search phrase is shown in the context and I am not forced to jump all its appearances.
![obrazek](https://github.com/sumatrapdfreader/sumatrapdf/assets/168529724/6631a52d-a989-4917-8c03-19b450384c22)

Thank you, SumatraPDF team, for the current work.

**Comment by g-berthiaume** [comment]
+1
This feature would not only be useful but also pair well with the "multi-document search" filter.

**Reply by GitHubRulesOK** [reply]
the Multi-document Search feature is simply allowing Microsoft Windows Explorer to read unsecured PDF contents, it is not MuPDF doing the search.

MS explorer search, even with Acrobats own filter, cannot read password locked contents, since Adobe say there are no backdoors to encrypted contents.

Many PDF files have naturally garbled contents even if it looks readable so search cannot find all files with a given word.
there are 2 identical copies of this PDF in the folder both with selectable text that says Abracadabra but it is not really serchable that way. **Only the filename** is a partial match.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/55c7a6fb-79b4-4e58-b43e-0b6a5496315b)

But if I search for **`9 letters`** with a single space then I find both ! NOTE the search is NOT case sensitive.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a681f455-e7e8-495e-8527-cf29715b2c91)

Note it does not matter how many spaces are used it is the 2 searchable strings are separated in any order
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/ac93528f-d441-432b-805a-6b72f515e202)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/fcc3a3de-ce2c-4360-9f82-71f7788dcf15)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/41eafa31-4cca-49e3-985b-eeb1dfaf9ea9)

Here you can see why search does not find `A b r a c a d a b r a`
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/6480ed92-3820-4937-9edb-83c2c4071988)

**Comment by g-berthiaume** [comment]
thanks for your answer @GitHubRulesOK.

I'm not a big fan of the Windows Explorer search. It's incredibly slow and forces you to use the file system structure to narrow your search.
To search multiple pdf, I recommend using [`rga`](https://github.com/phiresky/ripgrep-all).

All that said, having this feature integrated with sumatrapdf would be great for my pdf research workflow.

**Reply by GitHubRulesOK** [reply]
I never was a fan of Windows Version 4 search as exceptionally slow to store in index and cumbersome but now at version 11 it is way faster than any other ad hoc search tool I tried. The index for 30000+ files now takes seconds.

Windows search should be fast if restricted by configuration to those areas you need it, but as shown it has limitations compared to a commercial index app where you can add passwords to `pdftotext -layout` (resolves that abracadabra issue) or use regex queries on a stored list of all prior scanned contents.

So an app such as AgentRansack does those for docx pdf etc.
